### PR TITLE
Use uv to install dependencies in CI

### PR DESCRIPTION
**Pull request by danielhollas**
[uv](https://github.com/astral-sh/uv/issues/2155l) is the new cool kid in the town of Python packaging, from the creators of ruff. It currently serves as a much faster drop-in replacement for pip. 

For the `aiidalab` package, this drops installation time from ~15s to ~2s, which nearly halves the workflow time since the tests we have here are pretty quick.

Couple other small changes (feel free to push back)
 - Remove pre-commit section from CI since we use the pre-commit.ci anyway. 
 - Don't run tests on both push and pull_request event, this leads to duplication for pull requests coming from origin.
 - ~~remove bumpver and pre-commit from direct dev dependencies, as proposed [on Discourse](https://aiida.discourse.group/t/remove-pre-commit-and-bumpver-from-dev-dependencies/325)~~
 - get rid of incidental pyyaml dependency, see comment below

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -51,8 +51,6 @@ console_scripts =
 
 [options.extras_require]
 dev =
-    bumpver==2021.1114
```
Comment (danielhollas) [inline setup.cfg:54]: Unrelated change, I asked about this on Discourse.

File: tests/conftest.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 from pathlib import Path
 
 import pytest
-import yaml
+from ruamel.yaml import YAML
```
Comment (danielhollas) [inline tests/conftest.py:4]: We've been depending on `pyyaml` for tests by accident, and did not notice since it got installed together with pre-commit. Let's use ruamel since that's what is used in the aiidalab package.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -36,10 +28,11 @@ jobs:
               with:
                   python-version: ${{ matrix.python-version }}
 
-            - name: Install package
-              run: |
-                  pip install -e .[dev]
-                  pip freeze
+            - name: Install uv
+              run: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/astral-sh/uv/releases/download/0.1.21/uv-installer.sh | sh
```
Comment (danielhollas) [inline .github/workflows/ci.yml:32]:
```suggestion
              run: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/astral-sh/uv/releases/download/0.1.31/uv-installer.sh | sh
```

**Review by danielhollas — COMMENTED** [review]
@superstar54 could you take a look? This might be beneficial for QeApp as well.

File: setup.cfg (hunk 1 above)
Comment (unkcpz) [reply setup.cfg]: Hi @danielhollas, you add it back in the new commit correct?

Comment (danielhollas) [reply setup.cfg]: I did, since we didn't yet resolve this conversation on Discourse. (I still think it would be beneficial to remove, I'll reply on Discourse when I have time)

Comment (unkcpz) [reply setup.cfg]: Another option maybe don't pin down the version then if it already installed it will use it, otherwise install the latest version.

Comment (danielhollas) [reply setup.cfg:54]: I've unpinned them for now.

**Comment by danielhollas** [comment]
Going to merge this, we've got good experience with uv in AWB.
